### ISSUE TITLE
Debian 9 Stretch repositories and key

### ISSUE THREAD

**Issue by avra911** [chusiang/php7.ansible.role#17]
Hello,

This is not neccessary anymore in Debian 9 in the file `tasks/use-apt.yml`

```
- name: Add third-party repository for Debian
  become: true
  apt_repository: >
    repo="deb {{ debian_php7_apt_repo }} {{ ansible_distribution_release }} all"
    state=present
  when: ansible_distribution|lower == "debian"

- name: Add apt key for Debian
  become: true
  apt_key: url={{ debian_php7_apt_key }} state=present
  when: ansible_distribution|lower == "debian"

- name: Add third-party repository for Ubuntu
  become: true
  apt_repository: repo={{ ubuntu_php7_ppa_repo }}
  when: ansible_distribution|lower == "ubuntu"
```

Maybe it can use a new variable, default to `False`, called: `php_official_repo` and then use it in the when condition for the 3 tasks.

**Comment by chusiang** [comment]
Hi @avra911,

Maybe I can add a role of when like this.

if the **ansible_distribution = debian** and **ansible_distribution_major_version < 9**, it will ignore the tasks of `Add third-party repository for Debian` and `Add apt key for Debian`.

```
$ ansible server1 -m setup
        ...
        "ansible_distribution": "Debian", 
        "ansible_distribution_major_version": "8", 
        "ansible_distribution_release": "jessie", 
        "ansible_distribution_version": "8.8",
```

**Comment by avra911** [comment]
Hi, @chusiang!

That sounds definitely better solution.


Thanks for fast feedback!







---- On Wed, 21 Jun 2017 09:20:40 +0300 Chu-Siang Lai &lt;notifications@github.com&gt; wrote ----




Hi @avra911,

Maybe I can add a role of when like this.

if the ansible_distribution = debian and ansible_distribution_major_version = 9, it will ignore the tasks of Add third-party repository for Debian and Add apt key for Debian.

$ ansible server1 -m setup ... "ansible_distribution": "Debian", "ansible_distribution_major_version": "8", "ansible_distribution_release": "jessie", "ansible_distribution_version": "8.8",

**Comment by chusiang** [comment]
Hi @avra911 ,

I need to create new instance of Debian 9 for this feature, maybe I can finish it before this weekend.

Thanks for you feedback.

**Comment by chusiang** [comment]
It is working, now !

1. Check pass with Travis CI: https://travis-ci.org/chusiang/php7.ansible.role/builds/245437365
1. Release note: https://github.com/chusiang/php7.ansible.role/releases/tag/1.3.5

**Comment by chusiang** [comment]
Reopen and keep going ... :sleeping:

> https://github.com/chusiang/php7.ansible.role/commit/47423f5d72c893d46ec532faf3a3904789e6b5cf

**Comment by chusiang** [comment]
The http://packages.dotdeb.org of apt third-party repository didn't supported the Debian 9, I will add `when` roles for **Poka-yoke**.

```
$ cat tasks/use-apt.yml
...
- block:

  - name: Add third-party repository on Debian
    become: true
    apt_repository: >
      repo="deb {{ debian_php7_apt_repo }} {{ ansible_distribution_release }} all"
      state=present
    when:
      - ansible_distribution|lower == "debian"
      - ansible_distribution_major_version < "9"
  
  - name: Add apt key on Debian
    become: true
    apt_key: url={{ debian_php7_apt_key }} state=present
    when:
      - ansible_distribution|lower == "debian"
      - ansible_distribution_major_version < "9"
  
  - name: Add third-party repository on Ubuntu
    become: true
    apt_repository: repo={{ ubuntu_php7_ppa_repo }}
    when: 
      - ansible_distribution|lower == "ubuntu"

  when: apt_php_official_repo == false
...
```
 
That is mean if we set the `apt_php_official_repo == false`, we could not to add this third-party repository on Debian 9, now.

> Basically `php_official_repo` is almost all the time `False` (as Default), in rare case such as Debian 9 and php7.0 is `True` and maybe some late Ubuntu versions. I believe it makes sense to allow back third-party repositories in all versions and do a manual switch which allow all cases back and leave more flexibility.
>
> Also, thanks again for your efforts, I was able to deploy today based on the fixes, because I am using 7.0.x.
>
> Best regards and have a good night!

**Comment by chusiang** [comment]
I have added a variable `apt_php_third_party_repo`, and the default vaule is `true`.

You can use this feature after v1.3.7.

> https://github.com/chusiang/php7.ansible.role/releases/tag/1.3.7
